tc/app/helpers/create_dir_along_path

```diff
@@ -4,31 +4,28 @@
 
 source "$CLI_DIR/tc/framework/functions"
 
-while [[ $# -gt 0 ]]
-do
-key="$1"
+along_path=
+dir_name=
 
-case $key in
-    --along_path)
-    ALONG_PATH="$2"
+while echo "$1" | grep -q ^--; do
+  __key=${1//--/}
+  __key=${__key//-/_}
+  if [[ -n "$2" ]]; then
+    eval "${__key}"='$2'
     shift 2
-    ;;
-    --dir_name)
-    DIR_NAME="$2"
-    shift 2
-    ;;
-    *)
-    ;;
-esac
+  else
+    set "${__key}"
+    shift
+  fi
 done
 
 function _create_dir_along_path() {
   __along_path=$1
   __dir_name=$2
   if [ ! -d "$__along_path"/"$__dir_name" ]; then
-    mkdir "$_along_path"/"$__dir_name"
+    mkdir "$__along_path"/"$__dir_name"
```

Comment:
You don't know the path is available or not.

Reply:
did't understand

Reply:
you need to know if a directory `$__along_path` available or not.